mxd/minio.tf

```diff
@@ -0,0 +1,261 @@
+    labels = {
+      app = "minio-storage-claim"
+    }
+  }
+  wait_until_bound = false
+  spec {
+    access_modes = ["ReadWriteOnce"]
+    resources {
+      requests = {
+        storage = "20Gi"
+      }
+    }
+  }
+}
+
+resource "kubernetes_service" "minio-service" {
+  metadata {
+    namespace = kubernetes_namespace.minio.metadata.0.name
+    name      = "minio"
+  }
+  spec {
+    cluster_ip = "10.96.103.86"
+    port {
+      protocol    = "TCP"
+      name        = "minio-api"
+      port        = 9000
+      target_port = 9000
+    }
+    port {
+      name        = "minio-interface"
+      port        = 9001
+      target_port = 9001
+    }
+    selector = {
+      app = "minio"
+    }
+  }
+}
+
+resource "kubernetes_deployment" "minio" {
+  metadata {
+    namespace = kubernetes_namespace.minio.metadata.0.name
+    name      = "minio"
+    labels = {
+      app = "minio"
+    }
+  }
+
+  spec {
+    # replicas = 2
+    selector {
+      match_labels = {
+        app = "minio"
+      }
+    }
+    strategy {
+      type = "Recreate"
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "minio"
+        }
+      }
+
+      spec {
+        container {
+          image = "minio/minio:RELEASE.2022-03-17T06-34-49Z"
+          name  = "minio"
+          args  = ["server", "/storage", "--console-address=:9001"]
+          env {
+            name  = "MINIO_ROOT_USER"
+            value = local.minio-username
+          }
+          env {
+            name  = "MINIO_ROOT_PASSWORD"
+            value = local.minio-password
+          }
+          port {
+            container_port = 9000
+            host_port      = 9000
+          }
+          port {
+            container_port = 9001
+            host_port      = 9001
+          }
+          volume_mount {
+            name       = "storage"
+            mount_path = "/storage"
+            read_only  = false
+          }
+
+        }
+
+        volume {
+          name = "storage"
+          persistent_volume_claim {
+            claim_name = kubernetes_persistent_volume_claim.minio-pv-claim.metadata[0].name
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "kubernetes_job" "create_minio_bucket" {
+  depends_on = [kubernetes_deployment.minio, kubernetes_service.minio-service]
+  metadata {
+    name      = "create-minio-bucket"
+    namespace = kubernetes_namespace.minio.metadata[0].name
+  }
+  spec {
+    ttl_seconds_after_finished = "90"
+    completions                = 1
+    completion_mode            = "NonIndexed"
+    template {
+      metadata {
+        name = "create-minio-bucket"
+      }
+      spec {
+        container {
+          name    = "minio-client"
+          image   = "minio/mc:latest" # Use the appropriate MinIO client image
+          command = ["/bin/sh", "-c"]
+          args    = ["mc config host add minio http://${local.minio-url} ${local.minio-username} ${local.minio-password} && mc mb minio/alice-bucket"]
```

Comment:
Can we combine both bucket creation job in a single job. `mc mb minio/alice-bucket && mc mb minio/bob-bucket`

Reply:
As the Minio servers are now separated into their own modules, the task of creating buckets will occur within their respective modules.